test/all-templates.test.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+// This file contains Jest tests that run against every function template in
+// the repo.
+
+const path = require('path');
+const fs = require('fs');
+
+const projectRoot = path.resolve(__dirname, '..');
+// Assemble a list of template directories here, since templates.json
+// may not have all of them:
+const templates = fs
+      .readdirSync(projectRoot, { withFileTypes: true })
+      .filter(
+          (file) =>
+          file.isDirectory() &&
+              !file.name.startsWith(".") &&
+              !file.name.startsWith("_") &&
+              !["node_modules", "test", "coverage", "docs"].includes(file.name)
+      )
+      .map((dir) => dir.name)
+      .filter(
```

Comment:
It's probably because I'm doing the Advent of Code problems right now, but I'd prefer to see this filter squished into the previous one. I could even imagine a helper function here that would help in the test later too. Something like:

```js
const dirHasFile = (dir, file) => {
  try {
    const file = path.join(projectRoot, dir, file);
    fs.accessSync(file, fs.constants.F_OK);
    return true;
  } catch(_err) {
    return false;
  }
}
```

Then we can filter with:

```js
      .filter(
          (file) =>
          file.isDirectory() &&
              !file.name.startsWith(".") &&
              !file.name.startsWith("_") &&
              !["node_modules", "test", "coverage", "docs"].includes(file.name) &&
              dirHasFile(file.name, "package.json")
      )
```

And the test later can be:

```js
    it('should have a .env file', (done) => {
        expect(dirHasFile(template, '.env')).toBe(true);
    });
```

Reply:
But this is just an idea, don't get it in the way of merging.

Reply:
I like the DRYness of it though! And `all-templates.test.js` might get more use outside of this test one day. I'll refactor to this version.

Reply:
After discussing with @myfancypants, it looks like we'll omit the `package.json` check since my edge case might not occur for most developers; it's also fairly easy to work around in `git` if it does.